diff --git a/src/com/android/mms/service/MmsService.java b/src/com/android/mms/service/MmsService.java
--- a/src/com/android/mms/service/MmsService.java
+++ b/src/com/android/mms/service/MmsService.java
@@ -211,10 +211,11 @@ public class MmsService extends Service {
     private IMms.Stub mStub = new IMms.Stub() {
         @Override
-        public void sendMessage(int subId, int callingUser, String callingPkg,
-                Uri contentUri, String locationUrl, Bundle configOverrides,
-                PendingIntent sentIntent, long messageId, String attributionTag) {
+        public void sendMessage(int subId, String callingPkg,
+                Uri contentUri, String locationUrl, Bundle configOverrides,
+                PendingIntent sentIntent, long messageId, String attributionTag) {
             LogUtil.d("sendMessage " + formatCrossStackMessageId(messageId));
             enforceSystemUid();
+            int callingUser = UserHandle.getCallingUserId();

             MmsStats mmsStats = new MmsStats(MmsService.this,
                     mMmsMetricsCollector.getAtomsStorage(), subId, getTelephonyManager(subId),
@@ -288,16 +289,17 @@ public class MmsService extends Service {
         }

         @Override
-        public void downloadMessage(int subId, int callingUser, String callingPkg,
+        public void downloadMessage(int subId, String callingPkg,
                 String locationUrl, Uri contentUri, Bundle configOverrides,
                 PendingIntent downloadedIntent, long messageId, String attributionTag) {
             // If the subId is no longer active it could be caused by an MVNO using multiple
             // subIds, so we should try to download anyway.
             // TODO: Fail fast when downloading will fail (i.e. SIM swapped)
             LogUtil.d("downloadMessage: " + MmsHttpClient.redactUrlForNonVerbose(locationUrl) +
                     ", " + formatCrossStackMessageId(messageId));

             enforceSystemUid();
+            int callingUser = UserHandle.getCallingUserId();

             // Make sure the subId is correct
             if (!SubscriptionManager.isValidSubscriptionId(subId)) {
@@ -432,10 +434,11 @@ public class MmsService extends Service {
         }

         @Override
-        public Uri importMultimediaMessage(int callingUser, String callingPkg,
+        public Uri importMultimediaMessage(String callingPkg,
                 Uri contentUri, String messageId, long timestampSecs, boolean seen, boolean read) {
             LogUtil.d("importMultimediaMessage");
             enforceSystemUid();
+            int callingUser = UserHandle.getCallingUserId();
             return importMms(contentUri, messageId, timestampSecs, seen,
                 read, callingUser, callingPkg);
         }
@@ -526,9 +529,10 @@ public class MmsService extends Service {
         }

         @Override
-        public Uri addMultimediaMessageDraft(int callingUser,
-                String callingPkg, Uri contentUri) throws RemoteException {
+        public Uri addMultimediaMessageDraft(String callingPkg,
+                Uri contentUri) throws RemoteException {
             LogUtil.d("addMultimediaMessageDraft");
             enforceSystemUid();
+            int callingUser = UserHandle.getCallingUserId();
             return addMmsDraft(contentUri, callingUser, callingPkg);
         }
