diff --git a/core/java/android/database/sqlite/SQLiteTokenizer.java b/core/java/android/database/sqlite/SQLiteTokenizer.java
index 7e7c3fb9..0087f80c 100644
--- a/core/java/android/database/sqlite/SQLiteTokenizer.java
+++ b/core/java/android/database/sqlite/SQLiteTokenizer.java
@@ -69,6 +69,12 @@ public class SQLiteTokenizer {
      */
     public static final int OPTION_TOKEN_ONLY = 1 << 0;

+    /**
+     * Require that brackets in SQL are balanced; unbalanced brackets
+     * will result in an exception.
+     */
+    public static final int OPTION_CHECK_BRACKETS = 1 << 1;
+
     /**
      * Tokenize the given SQL, returning the list of each encountered token.
      *
@@ -90,6 +96,8 @@ public class SQLiteTokenizer {
         if (sql == null) {
             return;
         }
+        final boolean checkBrackets = (options & OPTION_CHECK_BRACKETS) != 0;
+        int bracketDepth = 0;
         int pos = 0;
         final int len = sql.length();
         while (pos < len) {
@@ -105,7 +113,7 @@ public class SQLiteTokenizer {
                 final int end = pos;

                 final String token = sql.substring(start, end);
-                checker.accept(token);
+                if (checker != null) checker.accept(token);

                 continue;
             }
@@ -142,7 +150,7 @@ public class SQLiteTokenizer {
                     } else {
                         token = tokenUnquoted;
                     }
-                    checker.accept(token);
+                    if (checker != null) checker.accept(token);
                 } else {
                     if ((options &= OPTION_TOKEN_ONLY) != 0) {
                         throw genException("Non-token detected", sql);
@@ -164,7 +172,7 @@ public class SQLiteTokenizer {

                 final String token = sql.substring(quoteStart + 1, quoteEnd);

-                checker.accept(token);
+                if (checker != null) checker.accept(token);
                 continue;
             }
             if ((options &= OPTION_TOKEN_ONLY) != 0) {
@@ -202,8 +210,20 @@ public class SQLiteTokenizer {
             // For this purpose, we can simply ignore other characters.
             // (Note it doesn't handle the X'' literal properly and reports this X as a token,
             // but that should be fine...)
+            if (checkBrackets) {
+                if (ch == '(') {
+                    bracketDepth++;
+                } else if (ch == ')') {
+                    if (--bracketDepth < 0) {
+                        throw genException("Unbalanced brackets", sql);
+                    }
+                }
+            }
             pos++;
         }
+        if (checkBrackets && bracketDepth != 0) {
+            throw genException("Unbalanced brackets", sql);
+        }
     }

     /**
